ranger/config/commands.py

```diff
@@ -1959,3 +1959,25 @@ def tab(self, tabnum):
             in sorted(self.modes.keys())
             if mode
         )
+
+
+class confirm(Command):
+    """:confirm <command_to_be_confirmed> + <potential command arguments>
+
+        Can be used with any command. Ranger will ask if it should execute the
+        given command.
+    """
+
+    def execute(self):
+
+        from functools import partial
+        command = self.arg(1)
```

Comment:
I don't like this as much. We should show the entire command *at least*. Ideally the command with macros resolved.

Reply:
Okay two questions here:
1. What do you mean by macros? I have done some searching through the codebse and on the Wiki here on Github, are macros just key bindings of commands? So if this is correct, you want to show which command exactly gets executed?

2. Shall I also show selection of files if you operate on commands that use a selection of files (e.g: delete or open_with)?

Reply:
Yes, selection's a good idea too. Though I think you should only show that if there's no arguments specified. In other words, always show just exactly what files it'll operate on.